### PR TITLE
[docs] Add a note on torchao/quanto benchmarks for CogVideoX and memory-efficient inference

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
# What does this PR do?

- Demonstrates serialization example with torchao (gist)
- int8 quantized inference
- benchmarks for different quantizations (gist)

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sayakpaul @stevhliu

### CODE REVIEW

File: docs/source/en/api/pipelines/cogvideox.md [hunk 1]
```diff
@@ -81,6 +81,59 @@ CogVideoX-2b requires about 19 GB of GPU memory to decode 49 frames (6 seconds o
   - With enabling cpu offloading and tiling, memory usage is `11 GB`
 - `pipe.vae.enable_slicing()`
 
+### Quantized inference
+
+[PytorchAO](https://github.com/pytorch/ao) and [Optimum-quanto](https://github.com/huggingface/optimum-quanto/) can be used to quantize the Text Encoder, Transformer and VAE modules to lower the memory requirement of CogVideoX. This makes it possible to run the model on free-tier T4 Colab or smaller VRAM GPUs as well! It is also worth noting that TorchAO quantization is fully compatible with torch.compile, which allows for much faster inference speed.
+
+```python
```
Comment (sayakpaul) [inline docs/source/en/api/pipelines/cogvideox.md:88]: Maybe we could just provide your gists here to keep the API specs clean and minimal?

File: docs/source/en/api/pipelines/cogvideox.md [hunk 2]
```diff
@@ -81,6 +81,14 @@ CogVideoX-2b requires about 19 GB of GPU memory to decode 49 frames (6 seconds o
   - With enabling cpu offloading and tiling, memory usage is `11 GB`
 - `pipe.vae.enable_slicing()`
 
+### Quantized inference
+
+[PytorchAO](https://github.com/pytorch/ao) and [Optimum-quanto](https://github.com/huggingface/optimum-quanto/) can be used to quantize the Text Encoder, Transformer and VAE modules to lower the memory requirement of CogVideoX. This makes it possible to run the model on a free-tier T4 Colab or lower VRAM GPUs as well!
```
Comment (stevhliu) [inline docs/source/en/api/pipelines/cogvideox.md:86]: 🔥 

```suggestion
[torchao](https://github.com/pytorch/ao) and [Optimum-quanto](https://github.com/huggingface/optimum-quanto/) can be used to quantize the text encoder, Transformer and VAE modules to lower the memory requirements. This makes it possible to run the model on a free-tier T4 Colab or lower VRAM GPUs!
```

File: docs/source/en/api/pipelines/cogvideox.md [hunk 3]
```diff
@@ -81,6 +81,14 @@ CogVideoX-2b requires about 19 GB of GPU memory to decode 49 frames (6 seconds o
...
+
+It is also worth noting that TorchAO quantization is fully compatible with `torch.compile`, which allows for much faster inference speed. Additionally, the models can be serialized and stored in a quantized datatype to save disk space when using PytorchAO. Find examples and benchmarks at these links:
```
Comment (stevhliu) [inline docs/source/en/api/pipelines/cogvideox.md:88]:
```suggestion
It is also worth noting that torch quantization is fully compatible with [torch.compile](./optimization/torch2.0#torchcompile), which allows for much faster inference speed. Additionally, models can be serialized and stored in a quantized datatype to save disk space with torchao. Find examples and benchmarks in the gists below.
```

**Review by stevhliu — APPROVED** [review]
Very nice, thank you!

File: docs/source/en/api/pipelines/cogvideox.md (hunk 3 above)
Comment (a-r-r-o-w) [reply docs/source/en/api/pipelines/cogvideox.md]: seems like the url here was incorrect as it isn't accessible. probably need to remove the `./` before `optimization`?

**Pull request merged**